File: src/main/java/com/raychenon/leetcode/RemoveDuplicatesSortedArray.java
package com.raychenon.leetcode;

/**
 * User: raychenon
 * Date: 2019-08-04
 * https://leetcode.com/problems/remove-duplicates-from-sorted-array/solution/
 */
public class RemoveDuplicatesSortedArray {

    public int removeDuplicates(int[] nums) {
        if (nums.length == 0) {
            return 0;
        }

        int count = 0;
        for (int index = 1; index < nums.length; index++) {
            if (nums[index] != nums[count]) {
                count = count + 1;
                nums[count] = nums[index];
            }
        }
        return count + 1;
    }
}

File: src/main/kotlin/com/raychenon/kotlin/leetcode/RemoveDuplicatesSortedArray.kt
package com.raychenon.kotlin.leetcode

/**
 * User: raychenon
 * Date: 2019-08-04
 * https://leetcode.com/problems/remove-duplicates-from-sorted-array/
 */
object RemoveDuplicatesSortedArray {

    /**
     * nums: pass by refence
     * Elements in array cannot be removed
     * Time complexity : O(n)
     * Space complexity : O(1)
     */
    fun removeDuplicates(nums: IntArray): Int {
        var count = 0
        if (nums.size == 0) return 0

        for (index in 0..nums.size - 1) {
            if (nums.get(index) != nums.get(count)) {
                count = count + 1
                nums.set(count, nums.get(index))
            }
        }
        return count + 1
    }
}
